xen-ocaml/build.sh

```diff
@@ -136,13 +136,21 @@ case $OCAMLOPT_VERSION in
   exit 1
   ;;
 esac
-# This directory doesn't really exist on 4.08, but it's also unneeded, so it's
-# effectively harmless to leave it here.
-CFLAGS="$CFLAGS -I../../byterun"
-cd otherlibs/bigarray && make CFLAGS="${CFLAGS} -I../unix -DIN_OCAML_BIGARRAY" ${BIGARRAY_OBJ}
-ar rcs ../../libxenotherlibs.a ${BIGARRAY_OBJ}
 
-cd ../../..
+case $OCAMLOPT_VERSION in
+4.04.2|4.05.*|4.06.*|4.07.*)
+  CFLAGS="$CFLAGS -I../../byterun"  cd otherlibs/bigarray && make CFLAGS="${CFLAGS} -I../unix -DIN_OCAML_BIGARRAY" ${BIGARRAY_OBJ}
+  ar rcs ../../libxenotherlibs.a ${BIGARRAY_OBJ}
+  cd ../../..
+  ;;
+4.08.*)
+  cp `ocamlfind query mmap`/libmmap_stubs.a libxenotherlibs.a
```

Comment:
@diml I think in `mirage-platform` we only need the stubs (mmap.o and mmap_ba.o). Is this the right way to do it, or is `libmmap_stubs.a` a name private to Dune?

Reply:
if you need these C functions in a xen-MirageOS-kernel, you'll need to cross-compile them with appropriate CFLAGS and includes. see https://github.com/mirage/checkseum / https://github.com/inhabitedtype/bigstringaf on how to do this.